crates/ra_hir_ty/src/lib.rs

```diff
@@ -680,6 +680,21 @@ impl Ty {
         }
     }
 
+    pub fn deref(&self) -> &Ty {
```

Comment:
for the name, how about `strip_references`? I'd expect a `deref` to also handle the `Deref` trait, and deref just one layer.

Reply:
Renamed to `strip_references` as suggested.